### ISSUE TITLE
Shielded Transfer Recipient should accept payment address

### ISSUE THREAD

**Issue by Rigorously** [anoma/namada-interface#1306]
When initiating a Shielded Transfer `Error: Unexpected Bech32m human-readable part znam, expected zsknam`.

zsknam refers to a spending/shielding key and should be kept secret. Not even namadaw with `--unsafe-show-secret` shows it. The recipient address field should accept a znam payment address instead.

![image](https://github.com/user-attachments/assets/c45aca2a-9abf-4418-b2c6-b671dd3407ba)

**Comment by euharrison** [comment]
It'll be done on https://github.com/anoma/namada-interface/pull/1191

**Comment by brentstone** [comment]
@euharrison Is this completely resolved following the merge of #1191?

**Comment by euharrison** [comment]
@brentstone I still need to test it to confirm

**Comment by Rigorously** [comment]
> @brentstone I still need to test it to confirm

Housefire-below with Namadillo v1.0.6 should have all features enabled https://interface.knowable.run/

The Bech32m error is gone. The Shielded Transfer form allows to enter a payment address now,

The transaction itself throws an error. It skips the signature and gets stuck at "Transfer to Namada Shielded".

![image](https://github.com/user-attachments/assets/394e7f84-f7b2-4048-8985-e26e69d6af5a)

![image](https://github.com/user-attachments/assets/25c8d20d-f9aa-4ea6-b4bc-3ba519e6a982)

**Comment by mateuszjasiuk** [comment]
Yeah this won't work for now as we introduced a change to unshielding/shielded transfers flow. We need to unify logic between MASP and Transfer tabs. I will let you know

**Comment by mateuszjasiuk** [comment]
https://github.com/anoma/namada-interface/pull/1433

**Comment by euharrison** [comment]
Tested against campfire testnet

- Transfer `shield -> shield` is working fine ✅ 
- Transfer `shield -> transparent` is returning this error below

![Screenshot 2024-12-30 at 16 33 13](https://github.com/user-attachments/assets/14c7c561-aa01-4705-8522-ce73a35a1f4d)

**Comment by brentstone** [comment]
@euharrison The tx is requiring more gas than the gas limit (which looks to be set by the interface prob?). Do we have functionality for a user to try higher gas? If not, this might be critically important now

**Comment by euharrison** [comment]
hmm weird, the json we are signing on keychain has the same gas as any other transfer. I'll explore this a little bit more, maybe this is something on the keychain or sdk side

**Comment by brentstone** [comment]
@euharrison I think @Fraccaman might know more, I remember him talking about gas limit choice in Namadillo at one point.

**Comment by brentstone** [comment]
I think that log is somewhat buggy, we should figure out why that large number is being returned and what it is exactly

**Comment by mateuszjasiuk** [comment]
So there are two problems:
- the gas limit hardcoded in the indexer is too low for unshielding and shielded transfers
- the displayed message is buggy

I've fixed the first part [here](https://github.com/anoma/namada-indexer/pull/244) by bumping gas limit in the indexer to 60_000.
The other part has to be fixed in the protocol code.

I think this issue can be closed and we can add issue for buggy error message in namada repo.
@brentstone wdyt?

**Comment by euharrison** [comment]
Shielded transfer tested now and it worked fine. Closing this issue.
In case we observe anything new, please open new issues
